### ISSUE TITLE
Add 'stdout-command' to 'dump' command

### ISSUE THREAD

**Issue by ozboss** [creativeprojects/resticprofile#153]
Please add a 'stdout-command' (analogous to 'stdin-command' flag of the backup command) to the 'dump' command. Together with #151 this would allow to restore for example some application database by simply calling `resticprofile application_db.dump latest` to restore the latest backup the application database.

**Comment by jkellerer** [comment]
Thats an interesting idea, though 'stdin-command' is primarily provided to allow scheduled backup of streams. Nevertheless I agree that it is more convenient when also the restore can be done with a single command call where only the snapshot ID (e.g. latest) has to be provided.

**Comment by ozboss** [comment]
Yes exactly, this would allow to restore the stream similarly to how it was backup up. My backup in this case is a mysqldump. Restic provides a good example for this exact [backup](https://restic.readthedocs.io/en/stable/040_backup.html#reading-data-from-stdin) and [restore](https://restic.readthedocs.io/en/stable/050_restore.html#printing-files-to-stdout) process.

**Comment by jkellerer** [comment]
I'll add `stdout-command` and `stdout-file` to all restic commands, so that their output can be redirected cleanly. This can not only benefit `dump` but also other restic commands that may generate JSON output. 

We may also add `stdout-is-binary: (true|false)` to control whether output is logged or printed when the `--quiet` flag isn't set (binary output makes no sense when it isn't clean, printing it may in text mode just cause failures). This would then default to `true` for `dump` (or any other command that generates binary data).

**Comment by damoclark** [comment]
> We may also add stdout-is-binary: (true|false) to control whether output is logged or printed when the --quiet flag isn't set (binary output makes no sense when it isn't clean, printing it along with other output may just cause failures). This would then default to true for dump (or any other command that generates binary data).

I just got tricked up with this, attempting to restore a filesystem image snapshot using resticprofile with dump command.  Was baffled when it overflowed the device file saying no space left on device.  Only when I inspected the output of the dump that I twigged that resticprofile output was at the start.  

D.
